### ISSUE TITLE
many-to-many relationship / index

### ISSUE THREAD

**Issue by jerzydziewierz** [typora/typora-issues#3956]
# TLDR

* Enable many-to-many linking through an old-book-style of "index", where ctrl-click on a keyword leads to a list of all occurrences of that keyword.
* Any word can be marked as a keyword, and there is an embeddable list of keywords similar to TOC

# Details

Thank you for this fantastic tool! I love Typora and I am now doing most of my professional work with it.

It occurred to me that the now-classic hyperlinks enable many-to-one relationship -- that is, from multiple places, you can get to a unique new place. They do not directly support many-to-many relationship like "This keyword is also mentioned in these places".

How about if we could:
1. Add any word in the document or folder of documents to a "watchlist" 
2. Browse the watchlist
3. From the watchlist, go to any place where this word appears

This is a little bit like "find all" functionality, but with a dedicated section where interesting words are listed straight up.
